onyo/lib/commands.py

```diff
@@ -379,59 +378,45 @@ def get(inventory: Inventory,
     if depth < 0:
         raise ValueError(f"-d, --depth must be 0 or larger, not '{depth}'")
 
-    # TODO: This removes duplicates AND returns pseudo-keys if `keys` is empty. Latter should be done by query.
-    #       Former superfluous - it's passed to query as a set anyways.
-    keys = sanitize_keys(keys, defaults=PSEUDO_KEYS)
-
-    filters = set_filters(
-        filter_strings, repo=inventory.repo,
-        rich=not machine_readable) if filter_strings else None
-
-    # TODO: This is once more convoluted. path limitation should be its own thing, not integrated in the query
-    #       Alternatively: path limiting could become just a filter. Implementation-wise that's easy, once pseudo-keys
-    #       are properly delivered by an Asset class and the path pseudo-key is implemented.
-    #       - This suggests a generic filter_assets method (for an Inventory)
-    #       - Gets filters, possibly arbitrary callables (see filter(callable, list) in get_assets_by_query)
-    #       - Check usecases for whether that can cover all the queries
-    results = get_assets_by_query(inventory.repo.asset_paths,
-                                  keys=set(keys),
-                                  paths=paths,
-                                  depth=depth,
-                                  filters=filters)
-    # TODO: Move this inside query. A returned asset (-dict) should be filled accordingly already.
-    #       See TODO for UNSET_VALUE definition.
-    results = fill_unset(results, keys, UNSET_VALUE)
-
-    # TODO: use `natsort` package.
+    keys = deduplicate(keys) or inventory.repo.get_required_asset_keys()
```

Comment:
IMO, we should should:

A) respect user intent, and do what they say
B) question the user, and warn/error

I'm comfortable with either of the above two behaviors (with a preference for A).

I don't like the (current) behavior:

C) silently do something different because we think we know better

Reply:
Agreed. I'd prefer A, too. So - will do.